src/libPersistence/ContractStorage2.cpp

```diff
@@ -698,6 +698,66 @@ void ContractStorage2::FetchStateDataForKey(map<string, bytes>& states,
   }
 }
 
+bool ContractStorage2::CheckIfKeyIsEmpty(const string& key, bool temp) {
+  std::map<std::string, bytes>::iterator p;
+  unordered_set<string> keys_to_be_deleted;
+
+  auto checkIfKeyIsToBeDeleted = [&](const string& key) mutable {
+    if (keys_to_be_deleted.find(key) != keys_to_be_deleted.end()) {
+      return true;
+    }
+    if (temp) {
+      if (t_indexToBeDeleted.find(key) != t_indexToBeDeleted.cend()) {
+        keys_to_be_deleted.insert(key);
+        return true;
+      }
+    }
+    if (m_indexToBeDeleted.find(key) != m_indexToBeDeleted.cend()) {
+      keys_to_be_deleted.insert(key);
+      return true;
+    }
+
+    return false;
+  };
+
+  if (temp) {
+    p = t_stateDataMap.lower_bound(key);
+    while (p != t_stateDataMap.end() &&
+           p->first.compare(0, key.size(), key) == 0) {
+      if (!checkIfKeyIsToBeDeleted(p->first)) {
+        return false;
+      }
+      ++p;
+    }
+  }
+
+  p = m_stateDataMap.lower_bound(key);
+  while (p != m_stateDataMap.end() &&
+         p->first.compare(0, key.size(), key) == 0) {
+    if (!checkIfKeyIsToBeDeleted(p->first)) {
+      return false;
+    }
+    ++p;
+  }
+
+  std::unique_ptr<leveldb::Iterator> it(
+      m_stateDataDB.GetDB()->NewIterator(leveldb::ReadOptions()));
+
+  it->Seek({key});
+  if (!it->Valid() || it->key().ToString().compare(0, key.size(), key) != 0) {
```

Comment:
@KaustubhShamshery , Can we shift this if statement on the top and make everything else as "else" statement.
If `  if (!it->Valid() || it->key().ToString().compare(0, key.size(), key) != 0)` is not `true` then we anyways returning true.It will save time for filling map `keys_to_be_deleted` which won't be required for this particular condition.

Reply:
Thanks @KaustubhShamshery for clearing doubts here. I understand now. We are checking in the map first before traversing through level DB.